timezone/src/modules/Timezone.rb

```diff
@@ -245,20 +251,25 @@ def get_zonemap
     # @return	the number of the region that contains the timezone
     #
     def Set(zone, really)
-      zmap = get_zonemap
+      # Do not update the timezone if it's forced and it was already set
+      if (Mode.installation || Mode.update) && readonly && !@timezone.empty?
```

Comment:
Yes, it tries to handle also the AutoYaST case. To be honest, initially I used `Stage.initial` but fortunately @schubi2 noticed that it could break AutoYaST.